anthropic-java-bedrock/src/main/kotlin/com/anthropic/bedrock/backends/BedrockBackendAdapter.kt

```diff
@@ -0,0 +1,592 @@
+                    } // For now, ignore any other path segments.
+                }
+            }
+            "complete" -> {
+                // Do nothing special.
+            }
+            else -> throw AnthropicException(
+                "Service is not supported for Bedrock: ${pathSegments[1]}.")
+        }
+
+        val json: ObjectNode? = bodyToJson(request.body, jsonMapper)
+        val modelId: String
+        val isStream: Boolean
+
+        if (json != null) {
+            val model = json.remove("model")
+                ?: throw AnthropicException("No model found in body.")
+
+            modelId = model.asText()
+            isStream = json.remove("stream")?.asBoolean(false) ?: false
+            json.put("anthropic_version", ANTHROPIC_VERSION)
+
+            val betaVersions = request.headers.values(HEADER_ANTHROPIC_BETA)
+                .flatMap { it.split(",") }.distinct()
+
+            if (betaVersions.isNotEmpty()) {
+                json.replace("anthropic_beta",
+                    jsonMapper.valueToTree(betaVersions))
+            }
+        } else {
+            throw AnthropicException("Request has no body")
+        }
+
+        return request.toBuilder()
+            .replaceAllPathSegments("model", modelId)
+            .addPathSegment(
+                if (isStream) "invoke-with-response-stream" else "invoke")
+            .body(jsonToBody(json, jsonMapper))
+            .build()
+    }
+
+    /**
+     * Signs the Amazon Bedrock request. Requests to AWS services are signed
+     * using the AWS secret access key (one of the required credentials). This
+     * operation adds new headers to the request that allow AWS to detect
+     * tampering or replay attacks.
+     *
+     * @param request The request to be signed. This will not be modified; it
+     *     will be copied and the modified copy will be returned.
+     *
+     * @return The signed request including the signature headers.
+     */
+    override fun signRequest(request: HttpRequest): HttpRequest {
+        val awsSignRequest = SdkHttpRequest.builder()
+            .uri(request.url)
+            .method(SdkHttpMethod.fromValue(request.method.toString()))
+            .apply {
+                // For the signature, copy the content type header from the body
+                // if the request object has no content type header.
+                if (request.headers.values(HEADER_CONTENT_TYPE).isEmpty()) {
+                    request.body?.contentType().let {
+                        appendHeader(HEADER_CONTENT_TYPE, it)
+                    }
+                }
+                request.headers.names().forEach { name ->
+                    request.headers.values(name).forEach { value ->
+                        appendHeader(name, value)
+                    }
+                }
+            }
+            .build()
+
+        val signer: AwsV4HttpSigner = AwsV4HttpSigner.create()
```

Comment:
I would just inline this as `AwsV4HttpSigner.create()` below; Not sure the variable is adding more value